pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -1026,41 +1070,97 @@ func registriesConfigIgnition(templateDir string, controllerConfig *mcfgv1.Contr
 	return &registriesIgn, nil
 }
 
-// getValidScopePolicies returns a map[scope]policyRequirement from ClusterImagePolicy
-func getValidScopePolicies(clusterImagePolicies []*apicfgv1alpha1.ClusterImagePolicy) (map[string]signature.PolicyRequirements, error) {
+// getValidScopePolicies returns a map[scope]policyRequirement from ClusterImagePolicy, a map[scope][namespace]policyRequirement from ImagePolicy CRs.
+// It skips ImagePolicy scopes that conflict with ClusterImagePolicy scopes and logs the conflicting scopes in the ImagePolicy Status.
+func getValidScopePolicies(clusterImagePolicies []*apicfgv1alpha1.ClusterImagePolicy, imagePolicies []*apicfgv1alpha1.ImagePolicy, ctrl *Controller) (map[string]signature.PolicyRequirements, map[string]map[string]signature.PolicyRequirements, error) {
 	clusterScopePolicies := make(map[string]signature.PolicyRequirements)
+	namespacePolicies := make(map[string]map[string]signature.PolicyRequirements)
+
 	for _, clusterImagePolicy := range clusterImagePolicies {
 		sigstoreSignedPolicyItem, err := policyItemFromSpec(clusterImagePolicy.Spec.Policy)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		for _, scope := range clusterImagePolicy.Spec.Scopes {
 			scopeStr := string(scope)
 			clusterScopePolicies[scopeStr] = append(clusterScopePolicies[scopeStr], sigstoreSignedPolicyItem)
 		}
 	}
 
-	return clusterScopePolicies, nil
+	for _, imagePolicy := range imagePolicies {
+		namespace := imagePolicy.ObjectMeta.Namespace
+		sigstoreSignedPolicyItem, err := policyItemFromSpec(imagePolicy.Spec.Policy)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		var conflictScopes []string
+	outerScope:
+		for _, scope := range imagePolicy.Spec.Scopes {
```

Comment:
Could you help add some explanations for what this loop is doing? I'm not sure I follow the logic

Reply:
Updated @yuqi-zhang PTAL